core/src/main/java/com/segment/analytics/kotlin/core/utilities/JSON.kt

```diff
@@ -83,6 +86,49 @@ fun JsonObject.getMapSet(key: String): Set<Map<String, Any>>? {
         null
 }
 
+// Utility function to apply key-mappings (deep traversal) and an optional value transform
+fun JsonObject.mapTransform(
+    keyMapper: Map<String, String>,
+    valueTransform: ((key: String, value: JsonElement) -> JsonElement)? = null
+): JsonObject = buildJsonObject {
+    val original = this@mapTransform
+    original.forEach { (key, value) ->
+        var newKey: String = key
+        var newVal: JsonElement = value
+        // does this key1 have a mapping?
+        keyMapper[key]?.let { mappedKey ->
+            newKey = mappedKey
+        }
+
+        // is this value a dictionary?
+        if (value is JsonObject) {
+            // if so, lets recurse...
+            newVal = value.mapTransform(keyMapper, valueTransform)
+        } else if (value is JsonArray) {
```

Comment:
it's probably better to have an extension method `mapTransform` for `JsonArray` and wrap this block of `else if` into that extension method for readability. otherwise `newVal` and `newValue` is a bit confusing.

Reply:
rest LGTM :)

Reply:
ooh good catch. will do